streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -577,34 +575,37 @@ private void runLoop() {
                     polledRecords = false;
                 }
 
-                // only record poll latency is long poll is required
-                if (longPoll) {
-                    streamsMetrics.pollTimeSensor.record(computeLatency(), timerStartedMs);
-                }
+                streamsMetrics.pollTimeSensor.record(computeLatency(), timerStartedMs);
             }
 
             // try to process several records from each task via the topology, and also trigger punctuate
             // functions if necessary, which may result in more records going through the topology in this loop
-            if (totalNumBuffered > 0 || polledRecords) {
-                totalNumBuffered = 0;
-
+            if (polledRecords) {
```

Comment:
Since we are always draining the buffer after the poll call, we can construct the list of tasks which should be a subset of the `activeTasks` during the above phase that have some records polled and added into them; then in this phase loop through this list of tasks and remove the task from the list whenever `task.process() == 0` and while loop until the list is empty.